### PR TITLE
US26136: Update in game notifications

### PR DESCRIPTION
**Pull request by jmargh**
Updating the in game notification system. This implements the notification system in lua and updates all core scripts that were using a GuiService:SendNotification() call. All new functionality is wrapped in fast flags.

### CODE REVIEW

**Comment by jmargh** [comment]
Dan doesn't like this.

File: CoreScriptsRoot/CoreScripts/NotificationScript2.lua [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+local EASE_STYLE = Enum.EasingStyle.Sine
+local TWEEN_TIME = 0.35
+local DEFAULT_NOTIFICATION_DURATION = 5
+
+--[[ Images ]]--
+local PLAYER_POINTS_IMG = 'http://www.roblox.com/asset?id=155363793'
+local BADGE_IMG = 'http://www.roblox.com/asset?id=177200377'
+local FRIEND_IMAGE = 'http://www.roblox.com/thumbs/avatar.ashx?userId='
+local GRAPHICS_QUALITY_LEVELS = 10
+local GRAPHICS_QUALITY_TO_INT = {
```
Comment (einsteinK) [inline CoreScriptsRoot/CoreScripts/NotificationScript2.lua]: Why not just use the Enum's .Value field?
For SavedQualitySetting it'll return the same value as predicted by this table.

File: CoreScriptsRoot/CoreScripts/NotificationScript2.lua [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+	end
+end
+LocalPlayer.FriendStatusChanged:connect(onFriendStatusChanged)
+
+--[[ Player Points Notifications ]]--
+local time = 20
+local function onPointsAwarded(userId, pointsAwarded, userBalanceInGame, userTotalBalance)
+	--if userId == LocalPlayer.userId then
+		if pointsAwarded == 1 then
+			sendNotifcation("Point Awarded", "You received "..tostring(time).." point!", PLAYER_POINTS_IMG, DEFAULT_NOTIFICATION_DURATION, nil)
```
Comment (einsteinK) [inline CoreScriptsRoot/CoreScripts/NotificationScript2.lua]: tostring(time) ?
Shouldn't it be "You received "..pointsAwarded.." point!" etc?
The local *time* seems to be useless...
(Except the tostring-case, it isn't used anywhere)

Comment (jmargh) [inline CoreScriptsRoot/CoreScripts/NotificationScript2.lua]: Yep, it was left in from a test case I was running, to test variable time notifications. Will be updated.

**Pull request closed without merging**